### PR TITLE
Fix for Issue# 1003

### PR DESCRIPTION
**Pull request by kandarp-n**
Fix for issue https://github.com/telefonicaid/iotagent-node-lib/issues/1003#issue-829138778

### CODE REVIEW

File: lib/services/ngsi/entities-NGSI-v1.js [hunk 1]
```diff
@@ -243,7 +243,7 @@ function sendUpdateValueNgsi1(entityName, attributes, typeInformation, token, ca
     if (
         'autoprovision' in typeInformation &&
         /* jshint -W101 */
-        typeInformation.autoprovision === undefined
+        typeInformation.autoprovision !== undefined
```
Comment (fgalan) [inline lib/services/ngsi/entities-NGSI-v1.js:246]: The same typo is not happening in lib/services/ngsi/entities-NGSI-v2.js file?

Just to be sure... :)

Comment (AlvaroVega) [reply lib/services/ngsi/entities-NGSI-v1.js:246]: It seems that it's ok:
https://github.com/kandarp-n/iotagent-node-lib/blob/8cc5aa4f5fcc8501d8e24e564ff425de91d814a6/lib/services/ngsi/entities-NGSI-v2.js#L403-L406

**Review by AlvaroVega — APPROVED** [review]
LGTM
CHANGES_NEXT_RELEASE file should be updated

**Comment by fgalan** [comment]
>  CHANGES_NEXT_RELEASE file should be updated with a new entry about this fix.

Yes. Please @kandarp-n include a line in the CHANGES_NEXT_RELEASE file about the fix done in this PR. It could be something like this:

```
- Fix: auto-provision functionality not working as expected when configured at device/group level in NGSIv1 API (#1003)
```

Anyway, you should contigure the IOTAs to use NGSIv2. The NGSIv1 API is deprecated and soon will be removed from IOTAs.

File: lib/services/ngsi/entities-NGSI-v1.js (hunk 1 above)
Comment (fgalan) [reply lib/services/ngsi/entities-NGSI-v1.js:246]: Great. NTC

**Comment by kandarp-n** [comment]
> > CHANGES_NEXT_RELEASE file should be updated with a new entry about this fix.
> 
> Yes. Please @kandarp-n include a line in the CHANGES_NEXT_RELEASE file about the fix done in this PR. It could be something like this:
> 
> ```
> - Fix: auto-provision functionality not working as expected when configured at device/group level in NGSIv1 API (#1003)
> ```
> 
> Anyway, you should contigure the IOTAs to use NGSIv2. The NGSIv1 API is deprecated and soon will be removed from IOTAs.

Hello @fgalan, is it ok to create a new commit for adding the line to CHANGES_NEXT_RELEASE file? or should I be modifying the existing commit?

**Comment by fgalan** [comment]
> Hello @fgalan, is it ok to create a new commit for adding the line to CHANGES_NEXT_RELEASE file? or should I be modifying the existing commit?

I think the simplest way is to add a new commit to kandarp-n:master branch with the change.

**Comment by fgalan** [comment]
The PR is perfect now to be merged :)

However, a little conflict in CHANGES_NEXT_RELEASE has appeared. Could you solve it, please? Thanks!

**Review by fgalan — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**